src/scripts/main.js

```diff
@@ -1,3 +1,53 @@
 'use strict';
 
-// write your code here
+const url = 'https://mate-academy.github.io/'
+  + 'phone-catalogue-static/api/phones.json';
+
+const detailsURL = 'https://mate-academy.github.io/'
+  + 'phone-catalogue-static/api/phones/';
+
+const list = document.createElement('ul');
+
+document.body.append(list);
+
+function getPhones() {
+  return fetch(url)
+    .then(response => {
+      if (!response.ok) {
+        return setTimeout(() => {
+          alert(`${response.status}`);
+        }, 5000);
+      }
+
+      return response.json();
+    });
+}
+
+getPhones()
+  .then(phones => {
+    const phonesIds = phones.map(phone => phone.id);
+
+    phones.forEach(el => {
+      list.insertAdjacentHTML('beforeend', `
+        <li>${el.name}</li>
+      `);
+    });
+
+    getPhonesDetails(phonesIds);
+  });
+
+function getPhonesDetails(array) {
+  const mappedArray = array.map(phoneId => {
+    return fetch(`${detailsURL}` + `${phoneId}.json`)
+      .then(response => response.json());
+  });
+
+  Promise.allSettled(mappedArray)
```

Comment:
```suggestion
  Promise.all(mappedArray)
```
![image](https://user-images.githubusercontent.com/38065883/132825587-432bea3d-4fb9-4cfd-8f0e-7876c3da730e.png)
all settled не реджектится на any error